lib/mockingjay/strategies/perfect_tree_traversal.ex

```diff
@@ -99,19 +99,7 @@ defmodule Mockingjay.Strategies.PerfectTreeTraversal do
     root_thresholds = Nx.flatten(thresholds[[.., 0]])
 
     {features, thresholds} =
-      Enum.reduce(1..(max_tree_depth - 1), {[], []}, fn depth, {all_nodes, all_biases} ->
-        start = @factor ** depth - 1
-        stop = @factor ** (depth + 1) - 2
-
-        n = Nx.flatten(features[[.., start..stop]])
-
-        b = Nx.flatten(thresholds[[.., start..stop]])
-
-        {[n | all_nodes], [b | all_biases]}
-      end)
-
-    features = Enum.reverse(features)
-    thresholds = Enum.reverse(thresholds)
+      _build_features_thresholds(features, thresholds, max_tree_depth: max_tree_depth)
```

Comment:
I believe this being inside a `def` we don't need to extract to a defn/deftransform. Returning `features` and `thresholds` as tuples should suffice